### PR TITLE
feat: Add session management

### PR DESCRIPTION
**Pull request by LucsT**
This PR add the CookieKonnector base to manage session.
Big diff because all lines are newly indented

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -9,188 +9,213 @@ moment.locale('fr')
...
+      return false
+    }
+  }
+
+  async fetch(fields) {
+    if (!(await this.testSession())) {
+      await this.logIn(fields)
+    }
+    const $ = await this.fetchPage()
+    const entries = await this.parsePage($)
```
Comment (doubleface) [inline src/index.js]: await not needed for this.parsePage if you remove async from parsePage method.

File: src/index.js [hunk 2]
```diff
@@ -9,188 +9,213 @@ moment.locale('fr')
...
+    return saveBills(entries, fields.folderPath, {
```
Comment (doubleface) [inline src/index.js]: this.saveBills or else this.request won't be used

File: src/index.js [hunk 3]
```diff
@@ -9,188 +9,213 @@ moment.locale('fr')
...
+      } else if (err.statusCode === 401) {
+        throw new Error(errors.LOGIN_FAILED)
+      } else {
+        throw new Error(errors.VENDOR_DOWN)
+      }
+    }
+    log('info', 'Successfully logged in.')
+  }
+
+  async fetchPage() {
```
Comment (doubleface) [inline src/index.js]: async to remove

File: src/index.js [hunk 4]
```diff
@@ -9,188 +9,213 @@ moment.locale('fr')
...
-  return entries
+    log('info', `Bill retrieved: ${entries.length} found`)
+    return entries
+  }
+
+  async getHistory() {
+    this.request = this.requestFactory({
+      json: false,
+      cheerio: true,
+      jar: true
```
Comment (doubleface) [inline src/index.js]: `jar: true` not needed. this.requestFactory adds this automatically

File: src/index.js (hunk 3 above)
Comment (LucsT) [reply src/index.js]: Async was remove from parsePage. this function need it as it contains request
Was just a swap mistake between fetchPage and parsePage

**Pull request merged**